analyzers/src/SonarAnalyzer.CSharp/Rules/StringOrIntegralTypesForIndexers.cs

```diff
@@ -34,10 +34,10 @@ namespace SonarAnalyzer.Rules.CSharp
     public sealed class StringOrIntegralTypesForIndexers : SonarDiagnosticAnalyzer
     {
         internal const string DiagnosticId = "S3876";
-        private const string MessageFormat = "Use string or an integral type here, or refactor this indexer into a method.";
+        private const string MessageFormat = "Use string, integral, index or a range type here, or refactor this indexer into a method.";
```

Comment:
If we change the message format in code we have to update it it on JIRA too.

Reply:
Maybe I would remove a from a range type? And  go with USe string, integral, index or range type here, ... WDYT?